parts/k8s/kuberneteswindowssetup.ps1

```diff
@@ -255,6 +255,13 @@ try
         Initialize-DataDirectories
 
         New-Item -ItemType Directory -Path "c:\k" -Force | Out-Null
+        icacls.exe "c:\k" /inheritance:r
+        icacls.exe "c:\k" /grant:r SYSTEM:`(OI`)`(CI`)`(F`)
+        icacls.exe "c:\k" /grant:r BUILTIN\Administrators:`(OI`)`(CI`)`(F`)
+        icacls.exe "c:\k" /grant:r BUILTIN\Users:`(OI`)`(CI`)`(RX`)

```

Comment:
I infer that the default permissions are less restrictive than what we're setting here.

Is allowing `NT AUTHORITY\Authenticated Users` a general security risk, or is there some specific scenario this addresses? Looks like it does what it says.

Reply:
Yup, (M) means that user/group can modify files.
kubelet, kube-proxy and other binaries are stored under c:\k so giving non-admins the ability to over-write those files could lead to an escalation of privilege attack.